CRM/Case/XMLProcessor/Process.php

```diff
@@ -848,32 +842,67 @@ private function getBoolSetting($settingKey, $xmlTag, $default = 0) {
 
   /**
    * At some point name and label got mixed up for case roles.
-   * Check for higher priority tag <machineName> first which represents name, then fall back to the <name> tag which somehow became label.
-   * We do this to avoid requiring people to update their xml files which can be stored in external files.
-   *
-   * Note this is different than doing something like comparing the <name> tag against name in the database and then falling back to comparing label in the database, which is subject to an edge case where you would get the wrong one (where the label of one relationship type is the same as the name of another). Here there are two tags with explicit single meanings.
+   * Check against known machine name values, and then if no match check
+   * against labels.
+   * This is subject to some edge cases, but we catch those with a system
+   * status check.
+   * We do this to avoid requiring people to update their xml files which can
+   * be stored in external files we can't/don't want to edit.
    *
    * @param SimpleXMLElement $xml
    *
-   * @return string
+   * @return array[bool|string,string]
    */
   public function locateNameOrLabel($xml) {
```

Comment:
Yes sorry I could have self-reviewed this part to help explain. The original idea of adding a new xml tag was based on something discussed during some rounds of cider, and I made the first PR towards it as a no-op that wouldn't take effect yet since the tag didn't exist yet, so it was still just doing what it did before. Then as I was clicking on more and more things I made a [comment](https://github.com/civicrm/civicrm-core/blob/d1a3ab5f2c73e6e48cb09ea8367302b2498d9e57/CRM/Case/BAO/CaseType.php#L188-L200) in one of the WIPs trying to convince myself ending with "hope this still makes sense to me later". Then a couple more clicks and reading it later, nope it didn't make sense, except for maybe one edge case. And then eventually I didn't need it all.